pdf_craft/sequence/generation.py

```
@@ -85,37 +89,24 @@ def _extract_page_references(jointer: Jointer) -> Generator[References, None, No
 
 def _page_index_from_layout(layout: AssetLayout | ParagraphLayout) -> int:
     if isinstance(layout, ParagraphLayout):
-        return layout.lines[0].page_index
+        return layout.blocks[0].page_index
     elif isinstance(layout, AssetLayout):
         return layout.page_index
     else:
         raise ValueError("Unknown layout type")
```

Ruff (0.14.8)

Prefer `TypeError` exception for invalid type

(TRY004)